src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -3065,7 +3070,7 @@ public void report(DiagnosticPosition _unused, JCDiagnostic details) {
         }
 
         void checkNewInnerClass(DiagnosticPosition pos, Env<AttrContext> env, Type type, boolean isSuper) {
-            boolean isLocal = type.tsym.owner.kind == MTH;
+            boolean isLocal = type.tsym.owner.kind == VAR || type.tsym.owner.kind == MTH;
```

Comment:
Most local classes will have their owner set to some method symbols. But, in some corner cases where a local class is defined inside a field initializer, the owner can actually be a *variable* symbol too.